users/utils/birthday_resolver.py

```diff
@@ -0,0 +1,136 @@
+import datetime
+
+import requests
+from django.conf import settings
+from helusers.user_utils import get_or_create_user
+from requests import RequestException
+from rest_framework import HTTP_HEADER_ENCODING
+from sentry_sdk import capture_exception
+
+
+def resolve_user(request, payload):
+    user = get_or_create_user(payload, oidc=True)
+
+    # If auth method is in configured loa levels we can try to read the date of birth of the user.
+    if (
+        payload.get("loa", "").lower()
+        in settings.OPEN_CITY_PROFILE_LEVELS_OF_ASSURANCES
+        and not user.date_of_birth
+    ):
+        b_day_reader = UserBirthdayReader(request)
+        try:
+            birthday = b_day_reader.get_user_birthday()
+        except (BirthDayReaderError, RequestException) as e:
+            capture_exception(e)
+        else:
+            user.date_of_birth = birthday
+            user.save()
+
+    return user
+
+
+class BirthDayReaderError(Exception):
+    pass
+
+
+class BirthDayReaderTokenNullOrEmptyError(BirthDayReaderError):
+    pass
+
+
+class BirthDayReaderQueryError(BirthDayReaderError):
+    pass
+
+
+class UserBirthdayReader:
+    """Reads birthday from national identification number stored in open city profile"""
+
+    CENTURY = {
+        "A": 2000,
+        "B": 2000,
+        "C": 2000,
+        "D": 2000,
+        "E": 2000,
+        "F": 2000,
+        "-": 1900,
+        "Y": 1900,
+        "X": 1900,
+        "W": 1900,
+        "V": 1900,
+        "U": 1900,
+    }
+
+    def __init__(self, request):
+        self.request = request
+
+    def __get_token(self):
+        token = self.request.headers.get("X-Authorization", b"")
+        if isinstance(token, str):
+            token = token.encode(HTTP_HEADER_ENCODING)
+        return token
+
+    def get_user_birthday(self) -> [datetime.date, None]:
+        nin = self.__get_national_identification_number()
+        if not nin or len(nin) != 11:
+            return None
```

Comment:
I'm wondering would be be a easy safety net for use log these cases 🤔 

Having an invalid length ID number and century look like corner cases we should never have, but if profile goes haywire, we just start to return `None`s without any information what is causing it. Unlikely, but if it happens, it could be a nightmare to debug and logs could help.

Reply:
Good point; we could add a warning message to sentry? Whatya think?

Reply:
Sounds great! 👍